torch/_inductor/decomposition.py

```diff
@@ -340,7 +340,32 @@ def add(x, y, *, alpha=None):
     if alpha is not None:
         z = alpha * y
     complex_type = torch.promote_types(x.dtype, y.dtype)
-    return (x.view(x.real.dtype) + z.view(y.real.dtype)).view(complex_type)
+    if x_is_complex_tensor and y_is_complex_tensor:
+        # For complex typed `x`, `x.view(x.real.dtype)` doubles the last dimension and can cause problem
+        # when broadcasting the add.
+        def reshape_tensor(tensor):
+            """Reshape tensor from [*initial_dims, last_dim] to *initial_dims, last_dim/2, 2]"""
+            # Get the current shape of the tensor
+            *initial_dims, last_dim = tensor.shape
+
+            # Check if the last dimension is even. We should never reach here since `x.view(x.real.dtype)`
+            # doubles the last dimension for complex numbers.
+            if last_dim % 2 != 0:
+                raise ValueError(
+                    "The size of the last dimension must be even to reshape it to [..., last_dim/2, 2]"
+                )
+
+            # Reshape the tensor
+            new_shape = (*initial_dims, last_dim // 2, 2)
+            reshaped_tensor = tensor.view(new_shape)
+            return reshaped_tensor
+
+        x_reshaped = reshape_tensor(x.view(x.real.dtype))
+        z_reshaped = reshape_tensor(z.view(y.real.dtype))
+        result = torch.flatten(x_reshaped + z_reshaped, start_dim=-2).view(complex_type)
+    else:
+        result = (x.view(x.real.dtype) + z.view(y.real.dtype)).view(complex_type)
```

Comment:
This is still wrong for `torch.randn(3) + torch.tensor([[1 + 1j]])`, right? (can we add a test for that case too?

Reply:
Oh I see what's going on